packages/esm-service-queues-app/src/queue-table/queue-entry-actions/queue-entry-actions.modal.tsx

```diff
@@ -287,89 +252,53 @@ export const QueueEntryActionModal: React.FC<QueueEntryActionModalProps> = ({
                   title={t('noPrioritiesConfigured', 'No priorities configured')}
                 />
               ) : (
-                <ContentSwitcher
-                  size="sm"
-                  selectedIndex={selectedPriorityIndex}
-                  onChange={(event) => {
-                    setSelectedPriorityUuid(event.name as string);
+                <RadioButtonGroup
+                  className={styles.radioButtonGroup}
+                  valueSelected={formState.selectedPriority}
+                  onChange={(uuid) => {
+                    setSelectedPriorityUuid(uuid);
                   }}>
                   {priorities?.map(({ uuid, display }) => (
-                    <Switch
-                      role="radio"
-                      name={uuid}
-                      text={
-                        uuid == queueEntry.priority.uuid
-                          ? t('currentValueFormatted', '{{value}} (Current)', {
-                              value: display,
-                              interpolation: { escapeValue: false },
-                            })
-                          : display
-                      }
+                    <RadioButton
                       key={uuid}
+                      name={display}
+                      labelText={
+                        <Tag
+                          className={`${priorities.findIndex((p) => p.uuid === uuid) === 1 && styles.orange} ${
+                            styles.tag
+                          }`}
+                          role="radio"
+                          key={uuid}
+                          value={uuid}
+                          type={(() => {
+                            const index = priorities.findIndex((p) => p.uuid === uuid);
+                            return index === 0 ? 'green' : index === 2 ? 'red' : '';
+                          })()}>
+                          {uuid == queueEntry.priority.uuid
+                            ? t('currentValueFormatted', '{{value}} (Current)', {
+                                value: display,
+                                interpolation: { escapeValue: false },
```

Comment:
Using `escapeValue` opts us out of `i18next` input sanitization and means we have to roll our own mechanism to mitigate against potential XSS attacks. See https://github.com/openmrs/openmrs-esm-patient-chart/pull/2230 for a possible alternative solution.